crates/virtio-queue/src/iterator.rs

```diff
@@ -131,10 +131,10 @@ mod tests {
             vq.desc_table().store(j, desc);
         }
 
-        vq.avail().ring().ref_at(0).store(0);
-        vq.avail().ring().ref_at(1).store(2);
-        vq.avail().ring().ref_at(2).store(5);
-        vq.avail().idx().store(3);
+        vq.avail().ring().ref_at(0).store(u16::to_le(0));
```

Comment:
Should we instead update the `store` and `load` from the mock interface to write and read little endian values so we don't need to use the conversion in all tests?

Reply:
I thought about that, the problem is that we don't always want to do the endianess translation. For instance, we don't want to do it for `Descriptor`, as it's already in the right endianess. Adding a new dedicated method doesn't solve the problem, as the dev would still need if the conversion needs to be done. I think the only "right" solution would be revamping the mock objects to avoid naked loads/stores from/to memory, having dedicated methods for each field with the right types, but that seems like too huge of a change to me.

Being pragmatic, as this only affects tests and it's probably going to be very hard to integrate a big-endian machine into the CI, I think it's fine if they get broken from time to time. I'll give them a try on each new virtiofsd-rs release and fix them if needed.

Reply:
Sounds good. We want to do some more improvements to the mock interface, do you mind opening an issue with this suggestion that you just mentioned?

Reply:
Sure, I've created #123 to track this. Thanks!